app/src/main/java/unipd/se18/ocrcamera/ManualTestOnSinglePhoto.java

```diff
@@ -0,0 +1,212 @@
+import android.graphics.Matrix;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import info.debatty.java.stringsimilarity.Damerau;
+import unipd.se18.ocrcamera.recognizer.OCR;
+import unipd.se18.ocrcamera.recognizer.OCRListener;
+import unipd.se18.ocrcamera.recognizer.TextRecognizer;
+
+import static unipd.se18.ocrcamera.recognizer.TextRecognizer.getTextRecognizer;
+
+/**
+ * Class that compare the image stored image in preferences with the same image rotate
+ * of a certain angle given by the user, showing the similarity
+ */
+
+public class ManualTestOnSinglePhoto extends AppCompatActivity {
+
+    private String startingOCRText;
+
+    /**
+     * Modification variable
+     */
+    private int angleRotation;
+
+    /**
+     * UI
+     */
+    private EditText degreeEditText;
+    private TextView degreeTextView;
+    private TextView confidenceTextView;
+    private TextView warningTextView;
+    private TextView foundTextView;
+
+    /**
+     * Listener used by the extraction process to notify results
+     */
+    private OCRListener ocrListener = new OCRListener() {
+        @Override
+        public void onTextRecognized(String text) {
+            /*
+             Text correctly recognized
+             Initialize Ui
+             Start compare OCRText of the modified photo with the one with no rotation
+             */
+            setContentView(R.layout.activity_manual_test_result);
+            degreeTextView = findViewById(R.id.degreeTextView);
+            confidenceTextView =findViewById(R.id.confidenceTextView);
+            warningTextView = findViewById(R.id.warningTextView);
+            foundTextView = findViewById(R.id.foundTextView);
+
+            SetResult(text);
+        }
+
+        @Override
+        public void onTextRecognizedError(int code) {
+            /*
+             Text not correctly recognized
+             Start CameraActivity to take a new photo
+             */
+            Intent i = new Intent(ManualTestOnSinglePhoto.this, CameraActivity.class);
+            startActivity(i);
+        }
+    };
+
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        //UI initialization
+        setContentView(R.layout.activity_manual_test);
+        degreeEditText = findViewById(R.id.editText1);
+
+
+        //Get image path and text of the last image from preferences
+        SharedPreferences prefs = this.getSharedPreferences("prefs", MODE_PRIVATE);
+        String pathImage = prefs.getString("imagePath", null);
+        startingOCRText = prefs.getString("text", null);
+
+        final Bitmap photo = BitmapFactory.decodeFile(pathImage);
+
+        Button fab = findViewById(R.id.btnConfirm);
```

Comment:
I suggest a better name for this button object like buttonAction or a better name for describing the object

Reply:
Yeah, it's the only button but anyway this would be better